diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c2e53cf..2c195de 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -122,7 +122,7 @@ if(NOT METAL_INCOMPATIBLE)
 endif()
 
 find_package(Threads REQUIRED)
-find_package(JsonCpp)
+find_package(jsoncpp CONFIG REQUIRED)
 
 ### All options defined here
 option(BUILD_LOADER "Build loader" ON)
diff --git a/src/loader/CMakeLists.txt b/src/loader/CMakeLists.txt
index 28aff53..6ee58f4 100644
--- a/src/loader/CMakeLists.txt
+++ b/src/loader/CMakeLists.txt
@@ -101,7 +101,11 @@ endif()
 
 # Get jsoncpp externally or internally
 if(BUILD_WITH_SYSTEM_JSONCPP)
-    target_link_libraries(openxr_loader PRIVATE JsonCpp::JsonCpp)
+    if(BUILD_SHARED_LIBS)
+        target_link_libraries(openxr_loader PRIVATE jsoncpp_lib)
+    else()
+        target_link_libraries(openxr_loader PRIVATE jsoncpp_static)
+    endif()
 else()
     if(NOT BUILD_LOADER_WITH_EXCEPTION_HANDLING)
         target_compile_definitions(openxr_loader PRIVATE JSON_USE_EXCEPTION=0)
