diff -r -u node/src/inspector/worker_inspector.h node.patched/src/inspector/worker_inspector.h
--- node/src/inspector/worker_inspector.h	2025-02-18 16:34:24.000000000 +0100
+++ node.patched/src/inspector/worker_inspector.h	2025-05-03 16:10:25.414466819 +0200
@@ -9,6 +9,7 @@
 #include <string>
 #include <unordered_map>
 #include <unordered_set>
+#include <cstdint>
 
 namespace node {
 namespace inspector {
